subprojects/common/file-storage/src/main/kotlin/com/avito/filestorage/HttpRemoteStorage.kt

```diff
@@ -1,178 +1,94 @@
 package com.avito.filestorage
 
-import com.avito.logger.LoggerFactory
-import com.avito.logger.create
-import com.avito.time.TimeProvider
+import com.avito.android.Result
 import okhttp3.HttpUrl
-import okhttp3.OkHttpClient
+import okhttp3.MediaType
+import okhttp3.MediaType.Companion.toMediaType
 import okhttp3.RequestBody.Companion.asRequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
 import retrofit2.Call
 import retrofit2.Callback
 import retrofit2.Response
+import java.io.File
 
-class HttpRemoteStorage(
+internal class HttpRemoteStorage(
     private val endpoint: HttpUrl,
-    httpClient: OkHttpClient,
-    private val timeProvider: TimeProvider,
-    loggerFactory: LoggerFactory
+    private val storageClient: FileStorageClient,
 ) : RemoteStorage {
 
-    private val logger = loggerFactory.create<HttpRemoteStorage>()
-
-    private val storageClient: FileStorageClient =
-        FileStorageClient.create(
-            endpoint = endpoint,
-            httpClient = httpClient
+    override fun upload(file: File, type: ContentType): FutureValue<Result<HttpUrl>> {
+        val mediaType = type.toMediaType()
+        return uploadInternal(
+            storageClient.upload(
+                content = file.asRequestBody(mediaType),
+                extension = type.toExtension()
+            )
         )
+    }
 
-    override fun fullUrl(result: RemoteStorage.Result.Success): String = "$endpoint${result.url}"
-
-    override fun upload(
-        uploadRequest: RemoteStorage.Request,
-        comment: String,
-        deleteOnUpload: Boolean
-    ): FutureValue<RemoteStorage.Result> {
-
-        val futureValue = SettableFutureValue<RemoteStorage.Result>()
+    override fun upload(content: String, type: ContentType): FutureValue<Result<HttpUrl>> {
+        val mediaType = type.toMediaType()
+        return uploadInternal(
+            storageClient.upload(
+                content = content.toRequestBody(mediaType),
+                extension = type.toExtension()
+            )
+        )
+    }
 
-        val timestamp = timeProvider.nowInSeconds()
+    private fun ContentType.toMediaType(): MediaType = when (this) {
+        ContentType.PNG -> "image/png"
+        ContentType.MP4 -> "video/mp4"
+        ContentType.TXT -> "text/plain"
+        ContentType.HTML -> "text/html"
```

Comment:
We force clients of `ContentType` duplicate such mappings.
Let's incorporate this knowledge into `ContentType` itself.

Reply:
done